src/App.tsx

```diff
@@ -1,155 +1,91 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useCallback, useEffect, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterType } from './types/FilterType';
 
-const USER_ID = 0;
+const USER_ID = 6704;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorText, setErrorText] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+
+  const fetchTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorText('Unable to load todos');
+      setTimeout(() => {
+        setErrorText('');
+      }, 3000);
+    }
+  }, []);
+
+  const closeError = () => {
+    setErrorText('');
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.All:
+          return todo;
+
+        case FilterType.Completed:
+          return todo.completed;
+
+        case FilterType.Active:
+          return !todo.completed;
+
+        default:
+          return todos;
+      }
```

Comment:
Use filter inside switch and not vice versa.
To avoid extra filter call in `ALL` case